Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs

```diff
@@ -1,54 +1,75 @@
 ﻿namespace Microsoft.ApplicationInsights.WindowsServer
 {
     using System;
-    using System.Threading;
-
     using Microsoft.ApplicationInsights.Channel;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.ApplicationInsights.Extensibility.Implementation;
     using Microsoft.ApplicationInsights.WindowsServer.Implementation;
-   
+
     /// <summary>
     /// A telemetry initializer that will gather Azure Web App Role Environment context information.
     /// </summary>
-    public class AzureWebAppRoleEnvironmentTelemetryInitializer : ITelemetryInitializer
+    public sealed class AzureWebAppRoleEnvironmentTelemetryInitializer : ITelemetryInitializer, IDisposable
     {
-        /// <summary>Azure Web App Hostname. This will include the deployment slot, but will be same across instances of same slot.</summary>
-        private const string WebAppHostNameEnvironmentVariable = "WEBSITE_HOSTNAME";
+        /// <summary>
+        /// Azure Web App Hostname. This will include the deployment slot, but will be 
+        /// same across instances of same slot. Marked as internal to support testing.
+        /// </summary>
+        internal string WebAppHostNameEnvironmentVariable = "WEBSITE_HOSTNAME";
 
         /// <summary>Predefined suffix for Azure Web App Hostname.</summary>
         private const string WebAppSuffix = ".azurewebsites.net";
 
+        /// <summary>
+        /// Value used for keeping track of when the hostname changes (slot swaps occur). We use this
+        /// to notify the other class that makes use of these environment variables.
+        /// </summary>
         private string nodeName;
         private string roleName;
 
+        private volatile bool updateEnvVars = true;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="AzureWebAppRoleEnvironmentTelemetryInitializer" /> class.
         /// </summary>
         public AzureWebAppRoleEnvironmentTelemetryInitializer()
         {
             WindowsServerEventSource.Log.TelemetryInitializerLoaded(this.GetType().FullName);
+            AppServiceEnvVarMonitor.Instance.MonitoredAppServiceEnvVarUpdatedEvent += this.UpdateEnvironmentValues;
         }
 
         /// <summary>
-        /// Initializes <see cref="ITelemetry" /> device context.
+        /// Initializes <see cref="ITelemetry" /> device context and helps keep any heartbeat values in sync as well.
```

Comment:
This initializer does not have any knowledge of heartbeat right? The comment look incorrect.

Reply:
Yes, thanks for catching this. Leftover from when I updated it last.